src/ex2/main.js

```
@@ -0,0 +1,67 @@
+class Main {
+  constructor() {}
+
+  init() {
+    const submitButton = domManager.getElement("list-item-submit");
+    submitButton.addEventListener("click", () => {
+      let input = domManager.getElement("list-item-input").value;
+
+      // number
+      if (/^[0-9]+$/.test(input)) {
+        pokemonClient.getPokemon(input).then((pokemon) => {
+          let item = pokemon.endsWith("was not found")
+            ? pokemon // error handling
+            : `catch ${pokemon}`; // real pokemon
+          this.addAndRenderItem(item);
+        });
+      }
+      // comma separated list of IDs
+      else if (/^[0-9, ]+$/.test(input)) {
+        pokemonClient
+          .getAllPokemons(input.split(",").map((e) => e.trim()))
+          .then((pokemons) => {
+            // error handling
+            if (typeof pokemons === "string") {
```

reading this code it is very counterintuitive. Why would the check here be that it is a string?
And then in this case that it is why is the response calling the addAndRenderItem without any string while in the case it's valid then you build the string here. 

I think it would be better to build both strings here to make the code more readable, or both in a different service. 
You could either throw an error when the pokemon is not found and then catch it here and build the right string or build the string of either not found or catch when you fetch the pokemon and then just add that string here. 
The way it is now it's a bit mixed between the responsibilities of these services